<a href="https://colab.research.google.com/github/angeliki-tzanou/datasci_5_statistics/blob/main/HHA_507_Assignment_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Assignment #5: Inderenital Statistics

##Chi-Square test:

- Loading Packages & dataset:



In [1]:
import pandas as pd

In [5]:
from scipy.stats import chi2_contingency as chi2

In [6]:
df_op = pd.read_csv('opiod.csv')

In [27]:
df_op.head(10)

Year  Patient County Code Patient County Name Rural/Urban       Payer  \
0  2013                  1.0              Albany       Urban    Medicaid   
1  2010                  1.0              Albany       Urban    Medicare   
2  2010                  2.0            Allegany       Rural    Medicaid   
3  2013                  2.0            Allegany       Rural    Medicaid   
4  2010                  2.0            Allegany       Rural       Other   
5  2012                  2.0            Allegany       Rural       Other   
6  2010                  3.0              Broome       Urban  Commercial   
7  2010                  3.0              Broome       Urban    Medicaid   
8  2010                  3.0              Broome       Urban    Medicare   
9  2010                  3.0              Broome       Urban       Other   

   ER Opioid  ER Visits   ER Rate Per 1000   Inpatient ER Opioid  \
0         22       35379              0.622                   15   
1         13       13486              0.964                   13   
2          0        3917              0.000                    2   
3          1        6273              0.159                    1   
4          1         451              2.217                    0   
5          0         560              0.000                    0   
6         24       24971              0.961                   13   
7         14       18615              0.752                   18   
8          6       12025              0.499                    4   
9          0        1103              0.000                    0   

   Inpatient ER Discharges  ...  Ambulatory Surgery Rate Per 1000  \
0                     3408  ...                               0.0   
1                    10785  ...                               0.0   
2                      422  ...                               0.0   
3                      292  ...                               0.0   
4                       59  ...                               0.0   
5                       62  ...                               0.0   
6                     2847  ...                               0.0   
7                     1793  ...                               0.0   
8                     6800  ...                               0.0   
9                      121  ...                               0.0   

   Outpatient Opioid  Outpatient Visits  Outpatient Rate Per 1000  \
0                  0              29976                       0.0   
1                  0                105                       0.0   
2                  0                 26                       0.0   
3                  0               5612                       0.0   
4                  0                 10                       0.0   
5                  0               1010                       0.0   
6                  0                 23                       0.0   
7                  0                 16                       0.0   
8                  0                 20                       0.0   
9                  0                  1                       0.0   

   Other Outpatient Total Opioid   Other Outpatient Total Discharges  \
0                              0                               33921   
1                              0                                9616   
2                              0                                 842   
3                              0                                6388   
4                              0                                 118   
5                              0                                1089   
6                              0                                9576   
7                              0                                1588   
8                              0                                6245   
9                              0                                 162   

   Other Outpatient Total Rate Per 1000  Overall Opioid  Overall Discharges  \
0                        

- Creating the table and the chi-square test:



In [28]:
df_op['Rural/Urban'].value_counts()

Urban    1196
Rural     718
State      36
Name: Rural/Urban, dtype: int64

In [29]:
df_op['Payer'].value_counts()

Medicaid      390
Medicare      390
Other         390
Commercial    390
Unknown       384
All Payers      6
Name: Payer, dtype: int64

In [7]:
contingency_table = pd.crosstab(df_op['Payer'], df_op['Rural/Urban'])

In [8]:
print(contingency_table)

Rural/Urban  Rural  State  Urban
Payer                           
All Payers       0      6      0
Commercial     144      6    240
Medicaid       144      6    240
Medicare       144      6    240
Other          144      6    240
Unknown        142      6    236


- Creating the chi-square test:

In [22]:
from scipy.stats.contingency import chi2_contingency

In [23]:
chi2, p, _, _= chi2_contingency(contingency_table)
print(f"Chi2 value: {chi2}")
print(f"P-value: {p}")

Chi2 value: 319.98611827610137
P-value: 9.183664413295381e-63


- Interpretation of results above:
In this case we can see that the Chi2 value and the P-value are substantially large. The chi2 value being aproximately around 320 signfies a strong correlation between the two variables. Where additionally the high P-value of about 9, represents a strong relationship between the variables of "Payer" & "Rural/Urban" rejecting the possibility of them being independent occurances, thus highlighting the that the type of payer and the area type of residence are stronly related.

